Update README, MIGRATION.md, JSDoc (#65)
Financial-Times/o-header-services: README.md

This header is for tools and services built by the Financial Times.

- [Markup](#markup)
	- [Title Section](#title-section)
	- [Primary navigation](#primary-navigation)
	- [Secondary navigation](#secondary-navigation)
	- [Themes](#themes)
	- [Bleed header](#bleed-header)
- [Sass](#sass)
- [JavaScript](#javascript)
- [Migration](#migration)
- [Contact](#contact)
- [License](#license)

## Markup

> An `o-header-services` header is divided into three main parts: title section, primary navigation and secondary navigation. In every case, this is the base of the variations in markup:

Reviewer: I found this section a little confusing. Maybe something like this is clearer:

```suggestion
An `o-header-services` header is divided into three main parts: title section _(required)_, primary navigation _(optional)_, and secondary navigation _(optional)_. Each is placed within a `header` element:
```

```html
<header class='o-header-services' data-o-component='o-header-services'>
	<!-- Markup specific to the needs of your product. Options detailed below. -->
</header>
```

### Title Section

A title section is **required** for every header, and every title section _must_ include a logo and a product title.
The logo will default to the FT logo—if you are not using the build service you can customise it with the `oHeaderServices()` [Sass mixin](#sass).

You can choose to include a product tagline to describe your product concisely, and there is space for extra content such as a 'Sign In' link in this section of the header, too.

_Note: If there is a 'Sign in' option, it should be consistent with other FT products by aligning itself to the right and avoiding different wording such as 'Log in' or 'Login'._

> You can see an example of a title section-only header and the markup for it in the [Origami Registry](https://registry.origami.ft.com/components/o-header-services#demo-title-section).

Reviewer: Attempting to cut the sentence length a little for readability 🔪 
```suggestion
For an example and markup, see the [title-only header in the Origami Registry](https://registry.origami.ft.com/components/o-header-services#demo-title-section).
```

### Primary Navigation

The primary navigation is an **optional** addition to the title section. It's most useful for high-level navigation.

This section of the header has specific behaviour, as it turns into a drawer at smaller viewport sizes (740px down). It requires the addition an extra element in the title section to support the hamburger button.

If you are using extra content (such as a 'Sign in' link), that will be pulled into the drawer, as well.

> You can see an example of the primary navigation and its markup in the [Origami Registry](https://registry.origami.ft.com/components/o-header-services#demo-primary-navigation).

Reviewer: 👇 Totally trivial, but putting "primary navigation" within the link makes the README more easy to scan.
```suggestion
You can see an example, with markup, of the [primary navigation in the Origami Registry](https://registry.origami.ft.com/components/o-header-services#demo-primary-navigation).
```

### Secondary Navigation

The secondary navigation is also an **optional** addition to the header, but it makes more sense alongside the primary navigation, as it serves more complicated products.

It includes two sections  of navigation, 'ancestors' and 'children'.
The 'ancestor' section  works in the form of a breadcrumb, and the children are relative to the ancestor.

At smaller viewports, it does _not_ collapse into the drawer, but becomes scrollable, instead.

> You can see an example of the secondary navigation and its markup in the [Origami Registry](https://registry.origami.ft.com/components/o-header-services#demo-secondary-navigation).

Reviewer: ```suggestion
You can see an example, with markup, of the [secondary navigation in the Origami Registry](https://registry.origami.ft.com/components/o-header-services#demo-secondary-navigation).
```

### Themes

> `o-header-services` offers theming for B2B or B2C products under FT.com. They are designed to affect the title section and the primary navigation.

Reviewer: ```suggestion
`o-header-services` offers theming for B2B or B2C products for FT.com products who use the master brand. They are designed to affect the title section and the primary navigation.
```

To add a theme to the header, add the appropriate class to the header element. For example, for B2B that would be:

```diff
+<header class='o-header-services o-header-services--b2b' data-o-component='o-header'>
-<header class='o-header-services' data-o-component='o-header'>
	<!-- Your header markup -->
</header>
```

> You can see what B2B and B2C headers look like on the [Origami Registry](https://registry.origami.ft.com/components/o-header-services#demo-theme-b2c).

Reviewer: ```suggestion
You can preview [B2B and B2C headers in the Origami Registry](https://registry.origami.ft.com/components/o-header-services#demo-theme-b2c).
```

### Bleed Header
If your application requires a bleed header, you'll need to add the `o-header-services--bleed` variant to your header.
```diff
+<header class='o-header-services o-header-services--bleed' data-o-component='o-header'>
-<header class='o-header-services' data-o-component='o-header'>
	<!-- Your header markup -->
</header>
```

## Sass

In order to output every type of `o-header-services` style, you'll need to include the following:
```scss
	@import 'o-header-services/main';

	@include oHeaderServices();
```

> You can be more selective about which types you would like to output, by using an `$opts` map. It accepts the following lists:

Reviewer: _logo isn't a list_
```suggestion
You can be more selective about which types you would like to output, by using an `$opts` map. It accepts the following options:
```

**types**
- `'primary-nav'`
- `'secondary-nav'`
- `'bleed'`
- `'b2b'`
- `'b2c'`

**logo**
- the name of a logo from the [logos image set](https://registry.origami.ft.com/components/logo-images@1.8.0). Defaults to the FT logo.

> If you are looking for specific `o-header-services` features you can request them in the following way:

Reviewer: ```suggestion
In this example we include only the styles for a [primary navigation](#primary-navigation) with the [bleed modifier](#bleed-header). We opt to use the Origami logo from the [logo image set](https://registry.origami.ft.com/components/logo-images@1.8.0).
```
